ambry-protocol/src/main/java/com/github/ambry/protocol/AmbryRequests.java

```diff
@@ -196,6 +196,77 @@ public void handleRequests(NetworkRequest networkRequest) throws InterruptedExce
+      DataInputStream dis = is instanceof DataInputStream ? (DataInputStream) is : new DataInputStream(is);
+      RequestOrResponseType type = RequestOrResponseType.values()[dis.readShort()];
+      Response response;
+      long requestQueueTime = SystemTime.getInstance().milliseconds() - request.getStartTimeInMs();
+      RequestMetricsUpdater metricsUpdater = new RequestMetricsUpdater(requestQueueTime, 0, 0, 0, true);
+      switch (type) {
+        case PutRequest:
+          PutRequest putRequest = PutRequest.readFrom(dis, clusterMap);
+          putRequest.accept(metricsUpdater);
+          response = new PutResponse(putRequest.getCorrelationId(), putRequest.getClientId(),
+              ServerErrorCode.Retry_After_Backoff);
+          break;
+        case GetRequest:
+          GetRequest getRequest = GetRequest.readFrom(dis, clusterMap);
+          getRequest.accept(metricsUpdater);
+          response = new GetResponse(getRequest.getCorrelationId(), getRequest.getClientId(),
+              ServerErrorCode.Retry_After_Backoff);
+
+          break;
+        case DeleteRequest:
+          DeleteRequest deleteRequest = DeleteRequest.readFrom(dis, clusterMap);
+          deleteRequest.accept(metricsUpdater);
+          response = new DeleteResponse(deleteRequest.getCorrelationId(), deleteRequest.getClientId(),
+              ServerErrorCode.Retry_After_Backoff);
+          break;
+        case TtlUpdateRequest:
+          TtlUpdateRequest ttlUpdateRequest = TtlUpdateRequest.readFrom(dis, clusterMap);
+          ttlUpdateRequest.accept(metricsUpdater);
+          response = new TtlUpdateResponse(ttlUpdateRequest.getCorrelationId(), ttlUpdateRequest.getClientId(),
+              ServerErrorCode.Retry_After_Backoff);
+          break;
+        case UndeleteRequest:
+          UndeleteRequest undeleteRequest = UndeleteRequest.readFrom(dis, clusterMap);
+          undeleteRequest.accept(metricsUpdater);
+          response = new UndeleteResponse(undeleteRequest.getCorrelationId(), undeleteRequest.getClientId(),
+              ServerErrorCode.Retry_After_Backoff);
+          break;
+        case ReplicaMetadataRequest:
+          ReplicaMetadataRequest replicaMetadataRequest =
+              ReplicaMetadataRequest.readFrom(dis, clusterMap, findTokenHelper);
+          replicaMetadataRequest.accept(metricsUpdater);
+          response = new ReplicaMetadataResponse(replicaMetadataRequest.getCorrelationId(),
+              replicaMetadataRequest.getClientId(), ServerErrorCode.Retry_After_Backoff,
+              ReplicaMetadataResponse.getCompatibleResponseVersion(replicaMetadataRequest.getVersionId()));
+          break;
+        case AdminRequest:
+          AdminRequest adminRequest = AdminRequest.readFrom(dis, clusterMap);
+          adminRequest.accept(metricsUpdater);
+          response = new AdminResponse(adminRequest.getCorrelationId(), adminRequest.getClientId(),
+              ServerErrorCode.Retry_After_Backoff);
```

Comment:
Retry_After_Backoff == 429 ?

Reply:
Yeah, it is an error code understood by Ambry frontends. We can treat it as similar to 429 in frontends and retry the requests.